### PR TITLE
Add configurable endpoint for Prometheus scraping

### PR DESCRIPTION
**Pull request by SimunKaracic**
Introduces new setting to keep backwards compatibility for kamon-prometheus endpoint

### CODE REVIEW

File: reporters/kamon-prometheus/src/main/resources/reference.conf [hunk 1]
```diff
@@ -82,6 +82,7 @@ kamon.prometheus {
     # Hostname and port used by the embedded web server to publish the scraping enpoint.
     hostname = "0.0.0.0"
     port = 9095
+    path = "/metrics"
```
Comment (Philippus) [inline reporters/kamon-prometheus/src/main/resources/reference.conf:85]: Maybe use `metrics-path` so it goes nicely with https://prometheus.io/docs/prometheus/latest/configuration/configuration/

File: reporters/kamon-prometheus/src/main/scala/kamon/prometheus/embeddedhttp/EmbeddedHttpServer.scala [hunk 2]
```diff
@@ -19,6 +19,6 @@ package kamon.prometheus.embeddedhttp
 import com.typesafe.config.Config
 import kamon.prometheus.ScrapeSource
 
-abstract class EmbeddedHttpServer(hostname: String, port: Int, scrapeSource: ScrapeSource, config: Config) {
+abstract class EmbeddedHttpServer(hostname: String, port: Int, path: String, scrapeSource: ScrapeSource, config: Config) {
```
Comment (Philippus) [inline reporters/kamon-prometheus/src/main/scala/kamon/prometheus/embeddedhttp/EmbeddedHttpServer.scala:22]: will this not break backwards compatibility for users that already defined their own `EmbeddedHttpServer` ?

Comment (SimunKaracic) [reply reporters/kamon-prometheus/src/main/scala/kamon/prometheus/embeddedhttp/EmbeddedHttpServer.scala]: I was not aware that people extended that! Fixed!

File: reporters/kamon-prometheus/src/main/resources/reference.conf (hunk 1 above)
Comment (SimunKaracic) [reply reporters/kamon-prometheus/src/main/resources/reference.conf:85]: Nice catch, will do!

Comment (Philippus) [reply reporters/kamon-prometheus/src/main/resources/reference.conf]: `metrics-path` is more encouraged HOCON style I think. Sorry to bother you. :)

Comment (SimunKaracic) [reply reporters/kamon-prometheus/src/main/resources/reference.conf]: Don't apologize, feedback is always welcome.

But I think I'm gonna stick with `_` on this one, Prometheus configuration uses metrics_path, so it makes sense if we're going for compatibility

File: reporters/kamon-prometheus/src/main/resources/reference.conf [hunk 3]
```diff
@@ -82,6 +82,7 @@ kamon.prometheus {
     # Hostname and port used by the embedded web server to publish the scraping enpoint.
     hostname = "0.0.0.0"
     port = 9095
+    metrics_path = "/metrics"
```
Comment (ivantopo) [inline reporters/kamon-prometheus/src/main/resources/reference.conf:85]: Please call it `metrics-path`, we don't user underscore on any of our setting names.

**Review by ivantopo — CHANGES_REQUESTED** [review]

**Pull request merged**